api/views.py

```diff
@@ -1,11 +1,34 @@
 from django.shortcuts import get_object_or_404
 from rest_framework import viewsets
-from rest_framework.response import Response
-
+from django.db.models import Q
+import math
 from api.serializers import *
 from customers.models import Provider
 
 
 class ProviderListViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = ProviderSerializer
     queryset = Provider.objects.all()
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        average_rating_min = self.request.query_params.get("average_rating_min", 0)
+        average_rating_max = self.request.query_params.get("average_rating_max", 5)
+
+        average_rating_min = int(average_rating_min)
+        average_rating_max = int(average_rating_max)
```

Comment:
You can cast to int directly at the assignment. 

Also what about floating values like min being 1.5 and max being 4.3 ?

```suggestion
        average_rating_min = int(self.request.query_params.get("average_rating_min", 0))
        average_rating_max = int(self.request.query_params.get("average_rating_max", 5))
```